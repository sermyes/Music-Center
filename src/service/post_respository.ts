import { FirebaseApp } from 'firebase/app';
import { getDatabase, Database } from 'firebase/database';

class PostRespository {
  private database: Database;
  constructor(private firebaseApp: FirebaseApp) {
    this.database = getDatabase(this.firebaseApp);
  }
}
export default PostRespository;
