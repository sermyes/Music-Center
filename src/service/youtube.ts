import { AxiosInstance } from 'axios';

class Youtube {
  constructor(private client: AxiosInstance) {}
  async playList() {
    const response = await this.client.get('playlistItems', {
      params: {
        part: 'snippet',
        playlistId: 'PL4fBVwST-63oKnpvGbt0Ok7Qkh3P22Plm',
        maxResults: 40
      }
    });

    return response.data.items;
  }
}

export default Youtube;
