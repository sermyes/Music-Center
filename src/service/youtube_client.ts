import axios from 'axios';

export const youtubeClient = axios.create({
  baseURL: 'https://youtube.googleapis.com/youtube/v3',
  params: { key: process.env.REACT_APP_YOUTUBE_API_KEY }
});
