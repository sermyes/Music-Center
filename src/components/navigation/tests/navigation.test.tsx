import Navigation from '../navigation';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { fakePosts } from './../../tests/posts';
import { PostData } from '../../../service/post_respository';

describe('Navigation', () => {
  const moveTo = jest.fn();
  const onActive = jest.fn();
  const updateIndex = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <Navigation
        onActive={onActive}
        moveTo={moveTo}
        updateIndex={updateIndex}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Check that the correct page index is entered when the button is clicked', async () => {
    render(
      <Navigation
        onActive={onActive}
        moveTo={moveTo}
        updateIndex={updateIndex}
      />
    );

    const item = screen.getAllByRole('listitem');

    await userEvent.click(item[0]);
    expect(moveTo).toHaveBeenCalledWith(0);

    await userEvent.click(item[1]);
    expect(moveTo).toHaveBeenCalledWith(1);

    await userEvent.click(item[2]);
    expect(moveTo).toHaveBeenCalledWith(2);

    await userEvent.click(item[3]);
    expect(moveTo).toHaveBeenCalledWith(3);
  });
});
