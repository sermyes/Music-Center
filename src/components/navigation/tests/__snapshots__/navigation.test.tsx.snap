// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<nav
  className="nav"
>
  <button
    className="btn"
    onClick={[Function]}
  >
    <i
      className="icon icon fas fa-bars"
    />
  </button>
  <div
    className="wrapper mui"
  >
    <ul
      className="ul"
    >
      <li
        className="menu menu"
        data-index="0"
        onClick={[Function]}
      >
        <i
          className="ico fas fa-home"
        />
        <span
          className="text muiText"
        >
          Home
        </span>
      </li>
      <li
        className="menu menu"
        data-index="1"
        onClick={[Function]}
      >
        <span
          className="text"
        >
          New
        </span>
      </li>
      <li
        className="menu menu"
        data-index="2"
        onClick={[Function]}
      >
        <i
          className="ico fas fa-music"
        />
        <span
          className="text muiText"
        >
          List
        </span>
      </li>
      <li
        className="menu menu"
        data-index="3"
        onClick={[Function]}
      >
        <i
          className="ico fas fa-comment"
        />
        <span
          className="text muiText"
        >
          Comment
        </span>
      </li>
    </ul>
  </div>
</nav>
`;
